content/4_Gold/PURS.mdx

```diff
@@ -314,14 +314,68 @@ public class MinSegmentTree {
 ```
 
 </JavaSection>
+
+<PySection>
+```python
+import sys
+inf = float('inf')
+input = sys.stdin.readline
+
+class SegmentTree:
+	def __init__(self, arr: list[int], n: int) -> None:
+		self.arr = arr
+		self.n = n
+		self.tree = [0] * (2 * n)
+
+		for ii in range(n):
+			self.tree[n + ii] = arr[ii]
+		for ii in reversed(range(n)):
+			self.tree[ii] = min(self.tree[ii << 1], self.tree[(ii << 1) | 1])
+
+	def update(self, i: int, v: int) -> None:
```

Comment:
add doc comments for `update` and `query` like in the module: https://usaco.guide/gold/PURS
also rename them to better describe what they do